analyzers/tests/SonarAnalyzer.Test/Trackers/ArgumentTrackerTest.cs

```diff
@@ -0,0 +1,1288 @@
+            context.Parameter.Name.Should().Be("value");
+            var method = context.Parameter.ContainingSymbol.Should().BeAssignableTo<IMethodSymbol>().Which;
+            method.MethodKind.Should().Be(MethodKind.PropertySet);
+            method.AssociatedSymbol.Should().BeAssignableTo<IPropertySymbol>().Which.Name.Should().Be(propertyName);
+        }
+    }
+
+    private static string WrapInMethodCS(string snippet) =>
+        $$"""
+        using System;
+        class C
+        {
+            public void M()
+            {
+                {{snippet}}
+            }
+        }
+        """;
+
+    private static string WrapInMethodVB(string snippet) =>
+        $$"""
+        Imports System
+        Class C
+            Public Sub M()
+                {{snippet}}
+            End Sub
+        End Class
+        """;
+
+    private static (SyntaxNode Node, SemanticModel Model) ArgumentAndModel(string snippet,
+        Func<string, MetadataReference[], (SyntaxTree Tree, SemanticModel Model)> compile, params Type[] argumentNodeTypes)
+    {
+        var pos = snippet.IndexOf("$$");
+        if (pos == -1)
+        {
+            throw new InvalidOperationException("The $$ maker was not found");
+        }
+        snippet = snippet.Replace("$$", string.Empty);
+        var (tree, model) = compile(snippet, MetadataReferenceFacade.SystemCollections.Concat(MetadataReferenceFacade.SystemDiagnosticsProcess).ToArray());
+        var node = tree.GetRoot().DescendantNodesAndSelf(new TextSpan(pos, 1)).Reverse().First(x => argumentNodeTypes.Any(t => t.IsInstanceOfType(x))); // root.Find does not work with OmittedArgument
+        return (node, model);
+    }
+
+    private static (SyntaxNode Node, SemanticModel Model) ArgumentAndModelCS(string snippet) =>
+        ArgumentAndModel(snippet, TestHelper.CompileCS, typeof(CS.ArgumentSyntax), typeof(CS.AttributeArgumentSyntax));
+
+    private static (SyntaxNode Node, SemanticModel Model) ArgumentAndModelVB(string snippet) =>
+        ArgumentAndModel(snippet, TestHelper.CompileVB, typeof(VB.ArgumentSyntax));
+
+    private static (bool Match, ArgumentContext Context) MatchArgumentCS(SemanticModel model, SyntaxNode node, ArgumentDescriptor descriptor) =>
+        MatchArgument<CSharpArgumentTracker, Microsoft.CodeAnalysis.CSharp.SyntaxKind>(model, node, descriptor);
+
+    private static (bool Match, ArgumentContext Context) MatchArgumentVB(SemanticModel model, SyntaxNode node, ArgumentDescriptor descriptor) =>
+        MatchArgument<VisualBasicArgumentTracker, Microsoft.CodeAnalysis.VisualBasic.SyntaxKind>(model, node, descriptor);
+
+    private static (bool Match, ArgumentContext Context) MatchArgument<TTracker, TSyntaxKind>(SemanticModel model, SyntaxNode node, ArgumentDescriptor descriptor)
+        where TTracker : ArgumentTracker<TSyntaxKind>, new()
+        where TSyntaxKind : struct
+    {
+        var context = new ArgumentContext(node, model);
```

Comment:
Why do we get as a parameter `SemanticModel model, SyntaxNode node` (which is, in fact, `ArgumentContext`) just to build it and return it? 

Since the caller already has this information it should not be returned. In this way, we can simplify the return type and return only the match.

```
    private static bool MatchArgument<TTracker, TSyntaxKind>(ArgumentContext context, ArgumentDescriptor descriptor)
        where TTracker : ArgumentTracker<TSyntaxKind>, new()
        where TSyntaxKind : struct =>
        new TTracker().MatchArgument(descriptor)(context);
```